docs/GETTING_STARTED.md

```diff
@@ -0,0 +1,239 @@
+```
+Behind the scenes soci is interacting with containerd.  So the tag
+`localhost:5000/rabbitmq` has to exist in containerd - that is what soci is
+referencing.  This created two kinds of objects.  The first is a series of ztocs
+(one per layer).  A ztoc is a table of contents for compressed data (z is the
+compression, toc is Table Of Contents).  The other is a manifest that relates
+the ztocs to the various layers.  Example output is : 
+```
+layer sha256:aa5c1807c64faef5411fdf8d572336478d2ae55881a348ca98d27de0c1031012 -> ztoc sha256:7482d8b46f52b9abc85b417e5cd6ce596fe078de428b46afad2943ec1ea1110c
+layer sha256:a2c7ff857687a23fe0c41f456c2b8f42359fbf35f3a5a6f3dc2cabee26aa4e9c -> ztoc sha256:4bbc8d97f9007e4023355f6a5634ad922d377256e94ccc322c975f19db351d3d
+layer sha256:08ed900bdfb06a0db4f07f002103a52dc480d98f448f939723cd49670e44a43d -> ztoc sha256:2d62d75c5342abfdab9acdeaa66f2f7d44032b70ac31469984da4e8bbb455605
+layer sha256:688ea927e54f8ad42dd715f3803db20f58f804970da295dbafa45ae8e094b588 -> ztoc sha256:d0864573c4166a91595bf19799d1e931ac15573ef4e535a61348f3afc7723a63
+layer sha256:11e66458f619be5b2e7ca677d6a82ea650682c6554487438e6f68ba3039248db -> ztoc sha256:50c548ae1967542cd50fd7796f20add555e24acbb03ea12608e3697c153afa01
+layer sha256:2f11156fa5ac4b97d38486ef618b292e112c7822d29b5dd7b425cb34a96b6594 -> ztoc sha256:d941838ee3b9862d0dfe4d0b50724818040584f0375a7c09257702d7fe40c9f5
+layer sha256:87d3a0863f984408afb6caf67a12832bc45cc3b3acbbf98c28ec916499c38a33 -> ztoc sha256:9e1c3ddb273db141e57fed0af6a0ba4a3b864adbf7bd42608715a6fb95a9e3b1
+layer sha256:3b65ec22a9e96affe680712973e88355927506aa3f792ff03330f3a3eb601a98 -> ztoc sha256:f9d786ee3e082fc671dac3e4b38dd1458a20e0425be31b6b09dfaa727925c3d2
+layer sha256:7e1cc2fa8c69560f02a99729c513ec7e3f49257d893bf8d30b5c6e7f50992644 -> ztoc sha256:4c1d63f476d4907e0db42b8736f578e79432a28d304935708c918c95e0e4df00
+```
+We can inspect one of these ztoc's with the following command (you will need to
+replace the digest with one of the ones created above) : 
+```
+sudo ./soci ztoc info sha256:4c1d63f476d4907e0db42b8736f578e79432a28d304935708c918c95e0e4df00
+```
+
+This will print to STDOUT the ztoc, which contains all of the information that
+soci needs to find a given file in the compressed tar.gz.  We can also view the
+index manifests by running : 
+
+```
+sudo ./soci index list
+```
+
+This will list out all of our index manifests.  To inspect an individual one we
+can use following (as always replace the digest with your own) : 
+
+```
+sudo ./soci index info sha256:f5f2a8558d0036c0a316638c5575607c01d1fa1588dbe56c6a5a7253e30ce107
+```
+
+This will dump out the index manifest in json.
+
+### Pushing the manifest to the registry
+Next we need to push the manifest to the registry with the following command.
+Just like with ctr we need to use `--plain-http` flag : 
+
+```
+sudo ./soci push --plain-http localhost:5000/rabbitmq:latest
+```
+
+This will push all of the ztoc files and the soci index manifest.
```

Comment:
Suggest emphasizing the whole and not its parts:
> This will push the SOCI index (the manifest all of the zTOCs) to the ORAS registry.

Reply:
Unrelated to this PR, but we should probably have these definitions somewhere prominent. We use the terms "SOCI index" and "SOCI index manifest" in various places and I know there has been some confusion about the distinction. Especially so since "index" is used to mean something else in the context of container images (i.e. a list of manifests)

Reply:
I really prefer being more explicit with things such as the parts vs the whole.  I think we should be writing this from the super user perspective asking - "what is going on here?" as opposed to the developer saying "Look how elegant this is".

Reply:
As per the definitions I've been trying to use "SOCI index manifest", but we definitely need a glossary.  How about we leave this now.  Do a glossary as a fast follow and update the doc as per the gloassary afterwards

Reply:
>  I think we should be writing this from the super user perspective asking - "what is going on here?" as opposed to the developer saying "Look how elegant this is".

That would be appropriate for a deep dive document (which would be a valuable addition to this repo), not a self-described *Getting Started* guide. After running `docker build`, you don't say "Now push the layers and image manifest to the remote registry". You say "Now push the image to the remote registry." Same thing [applies here](https://www.merriam-webster.com/dictionary/gestalt).

Reply:
Changed to 
> This will push all of the SOCI related artifacts.